src/SDK/MonitoringApplianceEmulator/Controls/TracesControl.xaml

```diff
@@ -0,0 +1,112 @@
+        <Grid.RowDefinitions>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="*"/>
+            <RowDefinition Height="Auto"/>
+        </Grid.RowDefinitions>
+        <Grid.ColumnDefinitions>
+            <ColumnDefinition Width="*"/>
+            <ColumnDefinition Width="Auto"/>
+        </Grid.ColumnDefinitions>
+        <StackPanel Orientation="Horizontal" HorizontalAlignment="Left" Grid.Row="0"  Grid.ColumnSpan="2">
+            <Label Content="Detector run:"/>
+            <ComboBox 
+                VerticalAlignment="Center" MinWidth="150" 
+                ItemsSource="{Binding Source={StaticResource LogNamesViewSource}}" SelectedItem="{Binding CurrentLogName}" >
+                <ComboBox.IsEnabled>
+                    <MultiBinding Converter="{StaticResource BooleanAndConverter}">
+                        <Binding Path="IsSmartDetectorRunning" Converter="{StaticResource InverseBooleanConverter}"/>
+                        <Binding Path="LoadLogTask.IsRunning" Converter="{StaticResource InverseBooleanConverter}"/>
+                    </MultiBinding>
+                </ComboBox.IsEnabled>
+            </ComboBox>
+        </StackPanel>
+        
+        <DataGrid
+            Grid.Row="1" Grid.Column="0" Grid.ColumnSpan="2" x:Name="TracesGrid"
+            ItemsSource="{Binding PageableLog.CurrentPageTraces}" AutoGenerateColumns="False" SelectionMode="Single">
+            <DataGrid.ItemContainerStyle>
+                <Style TargetType="{x:Type DataGridRow}">
+                    <Style.Triggers>
+                        <DataTrigger Binding="{Binding Level}" Value="Verbose">
+                            <Setter Property="Foreground" Value="DarkGray"/>
+                        </DataTrigger>
+                        <DataTrigger Binding="{Binding Level}" Value="Warning">
+                            <Setter Property="Background" Value="Orange"/>
+                        </DataTrigger>
+                        <DataTrigger Binding="{Binding Level}" Value="Error">
+                            <Setter Property="Background" Value="Red"/>
+                            <Setter Property="Foreground" Value="White"/>
+                        </DataTrigger>
+                    </Style.Triggers>
+                </Style>
+            </DataGrid.ItemContainerStyle>
+            <DataGrid.Columns>
+                <DataGridTextColumn
+                    Header="Time Stamp" Binding="{Binding TimeStamp, StringFormat='yyyy-MM-ddTHH:mm:ss.fff'}"
+                    Width="SizeToCells" IsReadOnly="True" CanUserSort="False" CanUserReorder="False"/>
+                <DataGridTextColumn
+                    Header="Level" Binding="{Binding Level}"
+                    Width="SizeToCells" IsReadOnly="True" CanUserSort="False" CanUserReorder="False"/>
+                <DataGridTextColumn
+                    Header="Message" Binding="{Binding Message}"
```

Comment:
>Binding Message [](start = 47, length = 15)

Trace messages can sometimes be very long (think about dump of results for later analysis). Can the message be trimmed here? The original message will still be in the log file, if needed. I'd say displaying a maximum of 10000 characters (per line) should be OK.

Reply:
talked it over

---
In reply to: [231003551](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/13#discussion_r231003551) [](ancestors = 231003551)